src/threading/MsgThread.cc

```diff
@@ -181,16 +228,17 @@ MsgThread::MsgThread() : BasicThread(), queue_in(this, nullptr), queue_out(nullp
     failed = false;
     thread_mgr->AddMsgThread(this);
 
-    if ( ! iosource_mgr->RegisterFd(flare.FD(), this) )
-        reporter->FatalError("Failed to register MsgThread fd with iosource_mgr");
-
-    SetClosed(false);
+    io_source = new detail::IOSource(this);
```

Comment:
What do you think about moving the `Register()` call out of the source ctor to over here? Otherwise this looks like a memory leak from the perspective of this class. Alternatively, at least put a comment here that the new`IOSource` is registering itself with the manager, which will take care of deleting it eventually.

Reply:
Moving Register() up is much better. Thanks.